### PR TITLE
Added AnalyzerRunner flag to Format all code files in solution

### PR DESCRIPTION
**Pull request by JoeRobich**
Added Format option to AnalyzerRunner. When set, all non-generated
C# and VB code files in the solution are formatted.

### CODE REVIEW

File: src/Tools/AnalyzerRunner/AnalyzerRunner.csproj [hunk 1]
```diff
@@ -11,6 +11,17 @@
     <!-- Automatically generate the necessary assembly binding redirects -->
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (sharwell) [inline src/Tools/AnalyzerRunner/AnalyzerRunner.csproj]: ❗️ Remove the `Condition` here so it applies to all builds

File: src/Tools/AnalyzerRunner/Program.cs [hunk 2]
```diff
@@ -14,9 +14,13 @@
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Diagnostics.Telemetry;
+using Microsoft.CodeAnalysis.Formatting;
 using Microsoft.CodeAnalysis.MSBuild;
+using Roslyn.Utilities;
 using File = System.IO.File;
 using Path = System.IO.Path;
+using CSharpSyntaxKind = Microsoft.CodeAnalysis.CSharp.SyntaxKind;
+using VBSyntaxKind = Microsoft.CodeAnalysis.VisualBasic.SyntaxKind;
```
Comment (sharwell) [inline src/Tools/AnalyzerRunner/Program.cs]: :bulb: `CSharp`/`VisualBasic` (both long form) or `CS`/`VB` (both short form)

File: src/Tools/AnalyzerRunner/Program.cs [hunk 3]
```diff
@@ -184,9 +188,65 @@ public static async Task Main(string[] args)
...
+                    if (GeneratedCodeUtilities.IsGeneratedCode(syntaxTree, isCommentTrivia, cancellationToken))
+                    {
+                        continue;
+                    }
+
+                    var documentOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+                    var formattedDocument = await Formatter.FormatAsync(document, documentOptions, cancellationToken).ConfigureAwait(false);
+
+                    var documentText = await formattedDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                    File.WriteAllText(formattedDocument.FilePath, documentText.ToString());
```
Comment (sharwell) [inline src/Tools/AnalyzerRunner/Program.cs]: @mavasani Is there a way to write the files back to disk by using a workspace operation?

Comment (JoeRobich) [reply src/Tools/AnalyzerRunner/Program.cs]: I updated to use `solution.WithDocumentText(document.Id, documentText)` and `solution.WithDocumentText(document.Id, documentText)`

File: src/Tools/AnalyzerRunner/Program.cs (hunk 2 above)
Comment (JoeRobich) [reply src/Tools/AnalyzerRunner/Program.cs]: done

File: src/Tools/AnalyzerRunner/AnalyzerRunner.csproj (hunk 1 above)
Comment (JoeRobich) [reply src/Tools/AnalyzerRunner/AnalyzerRunner.csproj]: sure thing

**Comment by jinujoseph** [comment]
test windows_debug_vs-integration_prtest

**Comment by jinujoseph** [comment]
test windows_debug_vs-integration_prtest

**Comment by JoeRobich** [comment]
Closing this PR as the work was moved to https://github.com/dotnet/roslyn/pull/29403

**Pull request closed without merging**